File: seed/pydantic-v2/examples/resources/types/types/node.py
from typing import List, Optional

from core.datetime_utils import serialize_datetime
from dt import datetime
from resources.types.types.node import Node
from resources.types.types.tree import Tree

from pydantic import BaseModel


class Node(BaseModel):
    name: str
    nodes: Optional[List[Node]] = None
    trees: Optional[List[Tree]] = None

    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}

File: seed/ruby-sdk-v2/examples/no-custom-config/lib/seed/types/types/node.rb
# frozen_string_literal: true

module FernExamples
  module Types
    module Types
      class Node < Internal::Types::Model
        field :name, -> { String }, optional: false, nullable: false
        field :nodes, -> { Internal::Types::Array[FernExamples::Types::Types::Node] }, optional: true, nullable: false
        field :trees, -> { Internal::Types::Array[FernExamples::Types::Types::Tree] }, optional: true, nullable: false
      end
    end
  end
end
